File: Data Structures & Algorithms/is-anagram/submission-0.go
func isAnagram(s string, t string) bool {
    if len([]rune(s)) != len([]rune(t)){
        return false
    }
    count := make(map[rune]int)
    for _, char := range s {
        count[char]++
    }
    for _, char := range t {
        count[char]--
        if count[char] < 0 {
            return false
        }
    }
    for _, val := range count {
        if val != 0 {
            return false
        }
    }
    return true
}

File: Data Structures & Algorithms/coin-change/submission-0.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        dp = [None] * (amount + 1)

        # Base case: 0 coins are needed to make amount 0
        dp[0] = 0

        for coin in coins:
            for money in range(1, amount + 1):
                prev = dp[money]
                if coin <= money and dp[money - coin] is not None:
                    cur = dp[money - coin] + 1
                    dp[money] = min(prev, cur) if prev is not None else cur

        return dp[amount] if dp[amount] is not None else -1